src/metabase/models/card.clj

```diff
@@ -655,12 +655,33 @@
       (pre-update-check-sandbox-constraints changes)
       (assert-valid-type (merge old-card-info changes)))))
 
+(defn- add-query-description-to-metric-card
+  "Add `:query_description` key to returned card.
+
+  Some users were missing description that was present in v1 metric API responses. This new key compensates for that.
+
+  This function is used in `t2/define-after-select :model/Card`. Metadata provider caching should be considered when
+  fetching multiple metric cards having common database, as done in eg. dashboard API context."
```

Comment:
Any particular concerns w.r.t. performance impact?

Reply:
I find it worth mentioning that if we find a context, where multiple metrics which share a database are loaded metadata provider caching could make this bit faster. That is currently relevant during dashboard load. Then, while a query is being processed. Metadata provider is also re-used there.